### PR TITLE
added 204 support to close EventSource

### PR DESCRIPTION
**Pull request by andresinaka**

### CODE REVIEW

File: EventSource/EventSource.swift [hunk 1]
```diff
@@ -81,7 +81,19 @@ public class EventSource: NSObject, NSURLSessionDataDelegate {
         readyState = EventSourceState.Closed
         urlSession?.invalidateAndCancel()
     }
-
+	
+	private func receivedMessageToClose(httpResponse: NSHTTPURLResponse?) -> Bool {
+		if httpResponse == nil {
```
Comment (volbap) [inline EventSource/EventSource.swift]: `guard let response = httpResponse else { return false }` and then you avoid the force-unwrap 😉

File: EventSource/EventSource.swift [hunk 2]
```diff
@@ -81,7 +81,19 @@ public class EventSource: NSObject, NSURLSessionDataDelegate {
...
+			return false
+		}
+		
+		if(httpResponse!.statusCode == 204) {
+			self.close();
```
Comment (volbap) [inline EventSource/EventSource.swift]: No semicolon;

File: EventSource/EventSource.swift [hunk 3]
```diff
@@ -108,7 +120,11 @@ public class EventSource: NSObject, NSURLSessionDataDelegate {
 //MARK: NSURLSessionDataDelegate
 
     public func URLSession(session: NSURLSession, dataTask: NSURLSessionDataTask, didReceiveData data: NSData) {
-        if readyState != EventSourceState.Open {
+		if receivedMessageToClose(dataTask.response as? NSHTTPURLResponse) {
```
Comment (volbap) [inline EventSource/EventSource.swift]: Please explicit `self.`

File: EventSource/EventSource.swift [hunk 4]
```diff
@@ -130,6 +150,10 @@ public class EventSource: NSObject, NSURLSessionDataDelegate {
 
     public func URLSession(session: NSURLSession, task: NSURLSessionTask, didCompleteWithError error: NSError?) {
         readyState = EventSourceState.Closed
+		
+		if receivedMessageToClose(task.response as? NSHTTPURLResponse) {
```
Comment (volbap) [inline EventSource/EventSource.swift]: Please explicit `self.`

File: EventSource/EventSource.swift (hunk 2 above)
Comment (andresinaka) [inline EventSource/EventSource.swift]: OMG how could this happen?!?!

**Pull request merged**